javalin/src/test/java/io/javalin/TestBeforeAfterMatched.kt

```diff
@@ -316,4 +317,38 @@ class TestBeforeAfterMatched {
             assertThat(slash.status).isEqualTo(HttpStatus.OK.code)
             assertThat(slash.headers.getFirst("X-After")).isEqualTo("true")
         }
+
+    @Test
+    fun `gh-2085 pathParamMap works with beforeMatched`() =
+        TestUtil.test(Javalin.create { config ->
+            config.router.apiBuilder {
+                get("/{path}") { ctx ->
+                    ctx.result(ctx.result() + "/" + ctx.pathParamMap())
+                }
+            }
+        }) { app, http ->
+            app.beforeMatched { ctx ->
+                ctx.result(ctx.pathParamMap().toString())
+            }
```

Comment:
I think this should be declared in config together with the `get` route.

They can both be rewritten to one-liners:

```
beforeMatched { it.result(ctx.pathParamMap().toString()) }
get("/{path}") { it.result(it.result() + "/" + it.pathParamMap()) }
```

Reply:
I basically copy pasted the example from the issue for testing. Will refine this.

Reply:
Oops, we forgot to add beforeMatched and afterMatched to the ApiBuilder :S